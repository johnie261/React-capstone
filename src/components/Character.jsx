import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Character = ({ image, name }) => (
  <Wrapper>
    <section>
      <li className="character">
        {image && <img src={image} alt={name} className="image" />}
        {!image && <h2 className="not-found">Image not found</h2>}
        <div className="txt">
          <p>{name}</p>
        </div>
      </li>
    </section>
  </Wrapper>
);

Character.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

const Wrapper = styled.div`
.character {
  list-style: none;
  .image {
    width: 100%;
    height: 60vh;
  }
  .txt {
    display: flex;
    justify-content: center;
    color: #fff;
    font-size: 1.5rem;
  }

}

@media only screen and (max-width: 1250px) {
  .character {
    .image {
      height: 40vh;
      width: 100%;
    }
  }

@media only screen and (max-width: 999px) {
  .character {
    .image {
      height: 25vh;
      width: 100%;
    }
  }
  .txt {
    font-size: 1.3rem;
  }
}
@media only screen and (max-width: 500px) {
  .character {
    .image {
      height: 29vh;
      width: 100%;
    }
  }
  .txt {
    p{
    font-size: 1.1rem;
    }
  }
}
`;

export default Character;
