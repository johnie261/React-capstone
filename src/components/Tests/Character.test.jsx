import React from 'react';
import { render } from '@testing-library/react';
import Character from '../Character';

describe('Characters', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Character
        name="name"
        image="image"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
