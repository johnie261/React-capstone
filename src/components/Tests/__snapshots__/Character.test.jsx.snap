// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB jtKaYx"
  >
    <li
      class="character"
    >
      <img
        alt="name"
        class="image"
        src="image"
      />
      <div
        class="txt"
      >
        <p>
          name
        </p>
      </div>
    </li>
  </div>
</DocumentFragment>
`;
